checks/handlerCheck.go

```diff
@@ -0,0 +1,96 @@
+package checks
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/newrelic/newrelic-lambda-extension/config"
+	"github.com/newrelic/newrelic-lambda-extension/lambda/extension/api"
+)
+
+type handlerRuntime func(handlerConfigs) error
+
+type handlerConfigs struct {
+	handlerName string
+	conf        *config.Configuration
+}
+
+var handlerCheck = map[string]handlerRuntime{
+	"node":   node,
+	"python": python,
+}
+
+var pathExists = func(p string) error {
+	if _, err := os.Stat(p); os.IsNotExist(err) {
+		return err
+	}
+	return nil
+}
+
+var checkAndReturnRuntime = func() handlerRuntime {
+	for k, v := range handlerCheck {
+		p := fmt.Sprintf("/var/lang/bin/%s", k)
+		err := pathExists(p)
+		if err == nil {
+			return v
+		}
+	}
+	// If we make it here that means the runtime is
+	// custom and we don't want to throw an error
+	return nil
+}
+
+var handlerPath = "/var/task"
+
+func checkHandler(conf *config.Configuration, reg *api.RegistrationResponse) error {
+	r := checkAndReturnRuntime()
+	if r != nil {
+		h := handlerConfigs{
+			handlerName: reg.Handler,
+			conf:        conf,
+		}
+		err := r(h)
+		if err != nil {
+			return fmt.Errorf("Lambda handler is set incorrectly: %s", err)
+		}
+	}
+	return nil
+}
+
+// Handler format for node functions: file.function
+// we can validate that the handler has been set to newrelic-lambda-wrapper.handler"
+// and that the file name provided in the env var NEW_RELIC_LAMBDA_HANDLER exists
+func node(h handlerConfigs) error {
+	functionHandler := getTrueHandler(h, "newrelic-lambda-wrapper.handler")
+	p := removePathMethodName(functionHandler)
+	p = pathFormatter(p, "js")
+	return pathExists(p)
+}
+
+// Handler format for python functions: file.function
+// we can validate that the handler has been set to newrelic_lambda_wrapper.handler
+// and that the file name provided in the env var NEW_RELIC_LAMBDA_HANDLER exists
+func python(h handlerConfigs) error {
+	functionHandler := getTrueHandler(h, "newrelic_lambda_wrapper.handler")
+	p := removePathMethodName(functionHandler)
+	p = pathFormatter(functionHandler, "py")
+	return pathExists(p)
+}
```

Comment:
This will work. But, I think we can do better, and there's an opportunity to play with idiomatic Go here. I'd suggest this:

Make an interface called `RuntimeHandlerCheck` with a method `check(h *handlerConfigs) error`

Make a struct `wrapperCheck`, and implement the RuntimeHandlerCheck interface. The name of the wrapper handler and the file extension should be fields in the struct.

Change `handlerCheck` to map strings to `RuntimeHandlerCheck` instances, and initialize it with appropriate values.

This eliminates code duplication, retains the flexibility of an arbitrary function call, and makes it clearer what is going on.

Reply:
In addition, ask yourself what other functions in this module are specific to the handler-wrapper case, and make them methods on the `wrapperCheck` struct, instead of top-level functions. 

That way, if we decide to do Java later or whatever, we'll have a clearer separation of concerns.